### PR TITLE
initial attempt for light group support

### PR DESCRIPTION
**Pull request by hzxa21**
Initial attempt for light group support:
- Add profile for light group and hue group
- Extract neccessary information in the "devicegroup" respoonse
- Construct the TerncyLight object for light group and hue group

I haven't tested with a real device group because I don't have time to mess up with my home devices and I don't have test devices. Maybe I will try next weekend.

related issue: #16 

## Future work:
- Set the availibility of the light group acoording to the light availibility
- Error handling in `async_turn_on` and `async_turn_off`

## Help wanted for the terncy team:
- Pass the group name back in the devicegroup response
- Pass the online state back in the devicegroup response

### CODE REVIEW

**Comment by hzxa21** [comment]
@rxwen This is a draft PR for light group support. Feel free to take a look.

**Review by rxwen — CHANGES_REQUESTED** [review]

File: terncy/const.py [hunk 1]
```diff
@@ -28,6 +28,8 @@
 PROFILE_LOCK = 11
 PROFILE_YANBUTTON = 6
 PROFILE_SWITCH = 4
+PROFILE_LIGHT_GROUP = 13
+PROFILE_HUE_GROUP = 12
```
[the remark this answers is not in the document]
Comment (hzxa21) [reply terncy/const.py]: fixed

**Comment by rxwen** [comment]
Thanks. LGTM.

**Comment by rxwen** [comment]
There is still a conflict in terncy/__init__.py. @hzxa21

**Comment by hzxa21** [comment]
> There is still a conflict in terncy/**init**.py. @hzxa21

Fixed.

I have several help wanted from your or terncy's side:
1. I still haven't get a chance to actually test it with a terncy gateway. Any chance you can help with that? In general, if we want to make this project easier to test, we can provide some kind of mock terncy server but this may leak terncy's API and protocols. Not sure whether this is suitable.
2. However, I do test it with homeassistent (with terncy being offline) and I found out one potential issue: switching on/off for a light-group always succeeds with this PR because we use async turn on/off and terncy doesn't give back the online flag for light-group responses (I also mention this in my PR description)/. Can terncy improves this?
3. light group name is also missing in the light-group response, which annoys me a bit because I need to manual rename the light group in homeassistant. Can terncy also improves this?

**Comment by rxwen** [comment]
> > There is still a conflict in terncy/**init**.py. @hzxa21
> 
> Fixed.
> 
> I have several help wanted from your or terncy's side:
> 
>     1. I still haven't get a chance to actually test it with a terncy gateway. Any chance you can help with that? In general, if we want to make this project easier to test, we can provide some kind of mock terncy server but this may leak terncy's API and protocols. Not sure whether this is suitable.

It's definitely helpful if the terncy system can be mocked for testing. To keep the API not leaked, an additional device management layer may need to be added so that the underlying communication details won't be required for testing. I'll consider about it.
And you don't have a Terncy home center right now?
> 
>     2. However, I do test it with homeassistent (with terncy being offline) and I found out one potential issue: switching on/off for a light-group always succeeds with this PR because we use async turn on/off and terncy doesn't give back the online flag for light-group responses (I also mention this in my PR description)/. Can terncy improves this?

As far as I know, a group doesn't have online flag. A group is a collection of devices that can be controlled as a whole. It is always online.
> 
>     3. light group name is also missing in the light-group response, which annoys me a bit because I need to manual rename the light group in homeassistant. Can terncy also improves this?

I think the Home Center can improve this in the coming version.

**Comment by hzxa21** [comment]
> > > There is still a conflict in terncy/**init**.py. @hzxa21
> > 
> > Fixed.
> > 
> > I have several help wanted from your or terncy's side:
> > 
> >     1. I still haven't get a chance to actually test it with a terncy gateway. Any chance you can help with that? In general, if we want to make this project easier to test, we can provide some kind of mock terncy server but this may leak terncy's API and protocols. Not sure whether this is suitable.
> 
> It's definitely helpful if the terncy system can be mocked for testing. To keep the API not leaked, an additional device management layer may need to be added so that the underlying communication details won't be required for testing. I'll consider about it.
> And you don't have a Terncy home center right now?
> > 
> >     2. However, I do test it with homeassistent (with terncy being offline) and I found out one potential issue: switching on/off for a light-group always succeeds with this PR because we use async turn on/off and terncy doesn't give back the online flag for light-group responses (I also mention this in my PR description)/. Can terncy improves this?
> 
> As far as I know, a group doesn't have online flag. A group is a collection of devices that can be controlled as a whole. It is always online.
> > 
> >     3. light group name is also missing in the light-group response, which annoys me a bit because I need to manual rename the light group in homeassistant. Can terncy also improves this?
> 
> I think the Home Center can improve this in the coming version.
> 

1. I have a home center in my new house but due to the covid situation in my city (guangzhou) i cannot go and get it... do you have a test home center that can be sent to me?

2. then even though my light goes offline, i can still set it on successfully on homeassitant. this sounds a bit counter-intuitive to me.

3. Thanks a lot!

**Pull request merged**